File: src/lib/types/graph.ts
// F-013 Project graph mirror types (matches src-tauri/src/models/graph.rs)

export type GraphNodeKind = 'repo' | 'project';
export type GraphEdgeKind = 'in_project' | 'cross_project_ms';

export interface GraphNode {
  id: string;
  label: string;
  kind: GraphNodeKind;
  role: string | null;
  repo_id: number | null;
  project_id: number | null;
}

export interface GraphEdge {
  source: string;
  target: string;
  kind: GraphEdgeKind;
}

export interface ProjectGraph {
  center: GraphNode | null;
  ring: GraphNode[];
  edges: GraphEdge[];
}

File: src-tauri/src/models/graph.rs
use serde::{Deserialize, Serialize};

// ── F-013 Project graph DTOs ──────────────────────────────────────────────

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum GraphNodeKind {
    Repo,
    Project,
}

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum GraphEdgeKind {
    InProject,
    CrossProjectMs,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct GraphNode {
    pub id: String,                  // "repo:42" or "project:7"
    pub label: String,               // display_name (repo) or project.name
    pub kind: GraphNodeKind,
    pub role: Option<String>,        // 'server' | 'client' | 'landing' | 'tool' | 'microservice' | None
    pub repo_id: Option<i64>,
    pub project_id: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct GraphEdge {
    pub source: String,
    pub target: String,
    pub kind: GraphEdgeKind,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ProjectGraph {
    pub center: Option<GraphNode>,
    pub ring: Vec<GraphNode>,
    pub edges: Vec<GraphEdge>,
}
